### PR TITLE
Fix visual prompt training for YOLOE26

### PR DESCRIPTION
**Pull request by ShuaiLYU**
### Summary
This PR fixes visual prompt training issues  YOLOE/YOLOE26 models.

#### 1. Add `tal_topk2` parameter support to TVPDetectLoss
- Added `tal_topk2` parameter to `TVPDetectLoss.__init__()` for flexible task-aligned assignment configuration
- Passes `tal_topk2` through to the underlying `v8DetectionLoss` for better one-to-one assignment control
- Removed unnecessary score slicing in `_get_vp_features` to use the full score tensor

#### 2. Fix typo in YOLOEDetect forward_lrpc
- Corrected `self.one2one_cv2` to `self.one2one_cv3` in the `forward_lrpc` method

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Fixes visual-prompt training behavior for YOLOe26 by correcting an end-to-end head routing bug and aligning TVP loss feature/parameter handling. 🛠️

### 📊 Key Changes
- Corrected `forward_lrpc()` to use `one2one_cv3` (not `one2one_cv2`) for classification features in end-to-end mode.
- Extended `TVPDetectLoss` to accept and forward an optional `tal_topk2` parameter into `v8DetectionLoss`.
- Updated VP feature extraction to use full `preds["scores"]` instead of slicing by `ori_nc`, preventing unintended channel dropping.

### 🎯 Purpose & Impact
- Prevents miswired heads during end-to-end training, improving correctness and stability for prompt-based detection. ✅
- Ensures TVP/visual-prompt loss uses the intended score tensor layout, avoiding shape/class mismatches. 🎯
- Improves configurability of task-aligned assignment via `tal_topk2`, helping advanced training setups behave as expected. ⚙️

### CODE REVIEW

**Comment by ShuaiLYU** [comment]
Test demo  @Laughing-q 

```
from ultralytics import YOLO
from ultralytics.models.yolo.yoloe import YOLOEVPTrainer

model=YOLO("yoloe-26s-seg.yaml").load("yoloe-26s-seg.pt")

data=dict(train=dict(yolo_data=["coco128.yaml"]),
          
          val=dict(yolo_data=["coco128.yaml"]))


model.train(
    data=data,
    task='detect',
    epochs=10,
    batch=4,
    imgsz=640,
    workers=4,
    single_cls=False,
    trainer=YOLOEVPTrainer)

```

**Comment by UltralyticsAssistant** [comment]
👋 Hello @ShuaiLYU, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! To ensure a seamless integration of your work, please review the following checklist:

-✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
-✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
-✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
-✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
-✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
-✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
-✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). This is an automated message, and an Ultralytics engineer will assist shortly. 🚀

**Review by UltralyticsAssistant — APPROVED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Looks clean. The changes are straightforward and align with the intended behavior; no issues found.

**Review by Laughing-q — APPROVED** [review]
@ShuaiLYU Nice fix!
